huaweicloud/services/servicestage/resource_huaweicloud_servicestage_repository_authorization.go

```diff
@@ -0,0 +1,174 @@
+package servicestage
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"regexp"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/chnsz/golangsdk/openstack/servicestage/v1/repositories"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+)
+
+func ResourceRepositoryAuth() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceRepositoryAuthCreate,
+		ReadContext:   resourceRepositoryAuthRead,
+		DeleteContext: resourceRepositoryAuthDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.All(
+					validation.StringMatch(regexp.MustCompile(`^[\w.-]*$`),
+						"The name can only contain letters, digits, underscores (_), hyphens (-) and dots (.)."),
+					validation.StringLenBetween(4, 63),
+				),
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"devcloud", "bitbucket", "github", "gitlab", "gitee",
+				}, false),
+			},
+			"host": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				RequiredWith: []string{"token"},
+				ExactlyOneOf: []string{"user_name"},
+			},
+			"token": {
```

Comment:
need to add `token` to `isSecurityFileds` function in config/logger to make it not appear in logs

Reply:
done.